packages/next/src/build/get-babel-loader-config.ts

```diff
@@ -24,11 +24,13 @@ const getReactCompilerPlugins = (
 
   const compilerOptions = typeof options === 'boolean' ? {} : options
   if (options) {
+    // TODO: Higher panic threshold in dev following https://react.dev/reference/react-compiler/panicThreshold#development-debugging
+    const defaultOptions = isDev ? {} : { panicThreshold: 'NONE' }
     return [
       [
         getReactCompiler(),
         {
-          panicThreshold: isDev ? undefined : 'NONE',
```

Comment:
This was actually not working as expected. `undefined` values are not permitted in loader options since `undefined` is not treated as serializeable. The serializability check is implemented with `deepEqual(options, JSON.parse(JSON.stringify(config)))`. We probably would consider an empty value the proper serialization for `undefined` but  handling that is more complicated so we just avoid `undefined` here.